editor/src/clj/editor/gl/vertex.clj

```diff
@@ -449,27 +448,29 @@ the `do-gl` macro from `editor.gl`."
 
 (defn- vertex-locate-attribs
   [^GL2 gl shader attribs]
-  (map
-    #(shader/get-attrib-location shader gl (name (first %)))
-    attribs))
-
-(defn- vertex-attrib-pointer
-  [^GL2 gl shader attrib stride offset]
-  (let [[nm sz tp & more] attrib
-        loc               (shader/get-attrib-location shader gl (name nm))
-        norm              (if (not (nil? (first more))) (first more) false)]
-    (when (not= -1 loc)
-      (gl/gl-vertex-attrib-pointer gl ^int loc ^int sz ^int (gl-types tp) ^boolean norm ^int stride ^long offset))))
+  (let [attribute-infos (shader/attribute-infos shader gl)]
+    (mapv (fn [attrib]
+            (let [attribute-name (name (first attrib))]
+              (if-some [attribute-info (get attribute-infos attribute-name)]
+                (:index attribute-info)
+                -1)))
+          attribs)))
```

Comment:
* Use the info we've already collected in `:attribute-infos` in functions that query shader info from the `ShaderLifecycle`.